source/features/update-pr-from-base-branch.tsx

```diff
@@ -104,7 +98,9 @@ async function init(): Promise<void | false> {
 	// - Draft PR (show the button anyway)
 	const canMerge = select.exists('[data-details-container=".js-merge-pr"]:not(:disabled)');
 	const isDraftPR = select.exists('[action$="ready_for_review"]');
-	if (!canMerge && !isDraftPR) {
+	const hasConflicts = select.exists('.js-merge-pr a[href$="/conflicts"]');
+
+	if ((!canMerge && !isDraftPR) || hasConflicts) {
```

Comment:
The logic overlaps, maybe the whole block should be be rewritten because the core logic is "if has conflicts or can not merge, return"

We already have `hasConflicts` but can we determine whether "it can merge" in a single selector? Without also checking specifically for draft PRs.

There are 8 situations to be tested, combinations of:

- PR state: regular or draft
- user: collaborator or third party
- conflicts: yes or no

Note: these are situations to be tested, not conditions that should appear in the code

Reply:
Why are we checking if its a draft PR?

Don't we only need to check if the user can merge?

Reply:
That's what I'm saying. We don't.

> the core logic is "if has conflicts or can not merge, return"

> Without also checking specifically for draft PRs.

Reply:
This doesn't work on draft PRs at all now because the button is always `.disabled` there